services/app.js

```diff
@@ -210,6 +223,79 @@ module.exports = function() {
             return deferred.promise;
         },
 
+        createAppKeys: function (appId, keyType,name){
+
+            console.log("Create App Masterkey function...");
+
+            var deferred = q.defer();
+            try {                
+
+                var promises = [];
+
+                console.log("Find AppID  exists or not...");
+                var collection =  global.mongoClient.db(global.keys.globalDb).collection("projects");
+                var findAppIdQuery = collection.find({appId:appId});
+                var findNameQuery  = collection.find({"keys.name":name})
```

Comment:
Why do we check name exist on ALL the apps. It should check name exist on that particular app.